### DISCUSSION TITLE
Images dont respect margins?

### DISCUSSION THREAD

**Discussion by joren-dev** [py-pdf/fpdf2#1155] [Q&A]
So I am using the `self.image` on a page, and while doing so I ran into the following problem
![image](https://github.com/py-pdf/fpdf2/assets/54151897/d0d7ab2e-6ceb-4967-a64a-65dad5151133)

It clearly does not respect the bottom margin that I've said like so in the ctor:
```py
margin_footer: int = 35
self.set_auto_page_break(True, margin_footer)
```

now the images are rendered as follows:
```py
def print_images(self, images, fixed_width, max_per_row=0, vertical_spacing=5):
    
    image_info = None
    images_in_row = 0
    
    # Iterate through images
    for image_path in images:
        # If maximum images per row is set and reached, move to the next row
        if max_per_row != 0 and images_in_row == max_per_row:
            self.set_x(self.l_margin)  # Reset x position
            self.set_y(self.get_y() + image_info.rendered_height + vertical_spacing)  # Move to next row
            images_in_row = 0  # Reset images in row counter

        # If adding this image would exceed the page width, move to the next row
        if self.get_x() + fixed_width > self.epw:
            self.set_x(self.l_margin)  # Reset x position
            self.set_y(self.get_y() + image_info.rendered_height + vertical_spacing)  # Move to next row
            images_in_row = 0  # Reset images in row counter

        # Display image
        image_info = self.image(image_path, x=self.get_x(), y=self.get_y(), w=fixed_width)

        # Update current_x and images_in_row for next iteration
        self.set_x(self.get_x() + fixed_width + 10)  # Add 10 units spacing between images
        images_in_row += 1

    # Set y to below images, reset x to left margin
    self.set_xy(
        self.l_margin,
        self.get_y() + (image_info.rendered_height if image_info else 0) + 5,
    )
```

Now for both the width and the height I make sure that itll fit within the margins, is this intended? I would think that for the bottom margin it would auto break like with `self.cell`, `self.tables`, etc.

I solved the issue myself by adding:
```py
# TODO: Figure out why images dont respect bottom margins
computed_image_height = calculate_image_height(image_path, fixed_width)

# Check if the image exceeds the page height
if self.get_y() + computed_image_height > self.page_break_trigger:
    self.add_page()  # Trigger page break
    self.set_xy(self.l_margin, self.t_margin)  # Reset x and y position
```
Prior to calling `image_info = self.image(image_path, x=current_x, y=current_y, w=fixed_width)
` which solved the issue.

![image](https://github.com/py-pdf/fpdf2/assets/54151897/2f9116ce-a9e2-45ae-ac5a-554a061b32f8)

However having to do something alike for each image would be pretty inefficient and I have a feeling I am doing something wrong which causes my behavior, could someone help me solve this mystery?

**Comment by joren-dev** [comment]
Not sure if this is the right place, but if it's not then please let me know and I'll move it elsewhere

**Reply by Lucas-C** [reply]
You did well!
I'm just going to convert this discussion into an issue , as it's a potential bug report: https://github.com/py-pdf/fpdf2/issues/1156

**Comment by joren-dev** [comment]
I've tried to make a minimal working example:

```py
from fpdf import FPDF

class PDF(FPDF):
    def __init__(self, orientation="P", unit="mm", format="A4"):
        super().__init__(orientation=orientation, unit=unit, format=format)
        self.set_auto_page_break(auto=True, margin=15)
        

# Create a new PDF object
pdf = PDF()

# Set up the document
pdf.set_title('My PDF Document')
pdf.set_author('John Doe')

# Add a chapter
pdf.add_page()
pdf.image('settings/meter_reading_example.png', w=100)
pdf.ln(2)
pdf.image('settings/meter_reading_example.png', w=100)
pdf.ln(2)
pdf.image('settings/meter_reading_example.png', w=100)
pdf.ln(2)
pdf.image('settings/meter_reading_example.png', w=100)
pdf.ln(2)
pdf.image('settings/meter_reading_example.png', w=100)
pdf.ln(2)
pdf.image('settings/meter_reading_example.png', w=100)
pdf.ln(2)
pdf.image('settings/meter_reading_example.png', w=100)
pdf.ln(2)
pdf.image('settings/meter_reading_example.png', w=100)


# Output the PDF
pdf.output('test.pdf')
```

But it seems to respect the margins just fine:
![image](https://github.com/py-pdf/fpdf2/assets/54151897/87bd2eef-dcf0-4a80-8325-070e943dde98)

So I am doing something wrong, just not sure what.

**Reply by Lucas-C** [reply]
You did right in trying to craft a minimal reproducing example!
It's not easy I guess, but without that it will be difficult for us maintainers to try to help you in solving your problem.
I'd b happy to have a look at it if you can provide some code that I can run 🙂

**Reply by joren-dev** [reply]
@Lucas-C I've posted a minimal reproducing example in the issue with the results.
